### DISCUSSION TITLE
Proper way to set up things in user directory

### DISCUSSION THREAD

**Discussion by marcinjahn** [blue-build/.github#22] [Support]
I wonder, what is the right way to set up things such as:

- my user account
- dotfiles
- homebrew packages
- any other files in `/home/{my-user}`

I imagine that I should be able to somehow specify the list of things to do (maybe a listing of files, or a git repo to clone to some location, or a script to execute) in my OS-image repository. On every update of the image, that list of things to do should get executed so that I would have my environment up to date not only on the first boot of a fresh intallation, but also after every update being installed (so, everyday).

I see that there's a [files module](https://blue-build.org/reference/modules/files/), but it specifically says that it's executed at build time of the image, so I suppose I should not use that one.

What is the proper way to do that? I'd like my fresh install of my OS to give me all the things I need OOTB, and to keep things up to date on every update. I want to minimize the necessary setup and have my OS just ready to use from the get go. Imagine I have 5 machines, and I just want all of them to be ready to use, all the time.

**Comment by xynydev** [comment]
We've floated the idea of creating a module for [adding files to the home directory](https://github.com/blue-build/modules/issues/118) and [running scripts on boot with systemd](https://github.com/blue-build/modules/issues/119), but nothing material exists yet.

You have the option of just writing to `/etc/skel/`, which is the home directory template for _new users_, and relying on manual copying of files for existing users. 

Or you could create a systemd unit that runs on boot and copies the files from say `/usr/share/my-os/homedir/` to the home directories of each user, whatever kind of structure that works. Would also need to figure out what to do when the process would overwrite user files etc. If you do that, we'd appreciate you documenting the journey, such that it could be used as the basis for an official homefiles module.

**Comment by SaltScratch** [comment]
For homebrew packages you could create a just recipe to install those. For dotfiles and home dir files take a look at the chezmoi module or copy them into /usr/etc/skel using the files module.

EDIT - looking at your last paragraph I think chezmoi may be the solution
